src/lib/trpc/routers/user.ts

```diff
@@ -212,44 +214,75 @@ export const userRouter = router({
 
 		return user?.Verification;
 	}),
-	resetPassword: publicProcedure.input(passwordResetSchema).query(async ({ input }) => {
-		const verification = await prisma.verification.findUnique({
-			where: { id: input.token }
-		});
-
-		if (!verification) {
-			throw new TRPCError({
-				message: `v-password.invalid-token`,
-				code: 'BAD_REQUEST'
+	validateVerificationToken: publicProcedure
+		.input(validateVerificationTokenSchema)
+		.query(async ({ input }) => {
```

Comment:
I think it is more valid o use `mutation` here, as it is manipulating data and not just retrieving it.
```suggestion
		.mutation(async ({ input }) => {
```